tox.ini

```diff
@@ -5,7 +5,7 @@
 
 [tox]
 isolated_build = true
-envlist = clean, py{38,39,310,311,312}-base, py{38,39,310,311,312}-grpc, report, docs
+envlist = clean, py{38,39,310,311,312}-base, py{38,39,310,311,312}-grpc, nicaiu, nicai_utf8, report, docs
```

Comment:
Please rename these to `py39-base-nicaiu` and `py39-base-nicai_utf8`. Two reasons:
- Be more explicit about which version we're testing with (`py39`).
- Inherit the `INSTALL_OPTS` and `PYTEST_OPTS` variables from `base`.

Note that stuff like `base:` and `nicaiu:` in the `setenv =` section doesn't have to match the entire env name. That way, we don't have to list out all of the combinations.